### PR TITLE
Issue 1992/emails in upcoming activities

### LINKED ISSUE #1992: Emails do not show up in "Upcoming activities"
## Description
The "Upcoming activities" section of the activities overview does not display emails. I think emails should be visible on the day that they are published.

## Steps to reproduce
1. Go to https://app.dev.zetkin.org/organize/1/projects
2. Create an empty project
3. Create an email using the "Create activity" button in top right
4. Configure email to be delivered tomorrow (requires adding some content, targeting the email etc)
5. Go back to the project overview

### Expected Behaviour
There should be an "Upcoming activities" sections with "today", "tomorrow" and "also this week", and the email should be listed as an activity in the "tomorrow" column.

### Actual Behaviour
Emails do not show up in the "Upcoming activities" section, so the project that only contains emails has no such section.

### Overview (with "Upcoming activities") in a project with only email
image

### Activities list for that same project
image

### Upcoming activities with surveys etc
image

## Proposed solution
I believe we just never got around to implementing this type of listing for emails. Have a look at how it works for tasks, surveys and call assignments. The difference for events is that they only have a single-day lifespan, the day they are sent. In theory they live (in peoples inboxes) forever after that, but I think it makes sense to only display them in one column, on the day that they are sent.

### PR DESCRIPTION
**Pull request by Jrende**
## Description
This PR adds email activities to upcoming activities in the overview

## Screenshots
upcomingemails

## Changes
* Adds emailActivities to the useActivityOverview hook

## Notes to reviewer
Also fixed a typo

## Related issues
Resolves #1992

### CODE REVIEW

**Review by ziggabyte — CHANGES_REQUESTED** [review]
Great work!! 

Even though the hope is that this would be as easy as a fix on two little lines, when I play with it today I'm seeing something that makes me think there is some work to be done in this area. 

My guess is that some logic regarding the send-time of the email is being miscalculated, making an email that was sent three hours today ago appear in both the column for today and the column for tomorrow: 
bild

I also think that the sentence at the bottom of the cards, that reads "Starts today" and "Ends in 21 hours" in this screenshot, is not optimal sentences for the emails. I think it should say "Will be sent tomorrow", "To be sent today" or "Was sent today" depending on the time of day. 

Would you like to investigate both the logic that makes the email appear in two columns at once, and the possibilities to customise the message at the bottom?

**Comment by Jrende** [comment]
> Great work!!
> 
> Even though the hope is that this would be as easy as a fix on two little lines, when I play with it today I'm seeing something that makes me think there is some work to be done...
> 
> My guess is that some logic regarding the use of the send-time of the email in the "tomorrow" column is wrong, making an email that was sent three hours ago today appear in both the column for today and the column for tomorrow: bild
> 
> I also think that the sentence at the bottom of the cards, that reads "Starts today" and "Ends in 21 hours" in this screenshot, is not optimal sentences for the emails. I think it should say "Will be sent tomorrow", "To be sent today" (or maybe "To be sent in xx hours"?) or "Was sent today" (or "Was sent xx hours ago"?) depending on the time of day.
> 
> Would you like to investigate both the logic that makes the email appear in two columns at once, and the possibilities to customise the message at the bottom?

So it was actually a medium-length task ;)
I can keep working on it to fix those issues.

**Comment by Jrende** [comment]
Screenshot_2024-06-08_11-13-51
I've customized the message for email activities and made sure that they are only visible for one day

File: src/utils/dateUtils.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export function isInFuture(datestring: string): boolean {
 }
 
 export function isSameDate(first: Date, second: Date): boolean {
-  return first.toISOString().slice(0, 10) == second.toISOString().slice(0, 10);
+  return dayjs(first).isSame(dayjs(second), 'day')
```
Comment (Jrende) [inline src/utils/dateUtils.ts:89]: This change might look dangerous. The reason I did it was that doing 
`(new Date("2024-06-08T00:00:00").toISOString().slice(0, 10))` returns "2024-06-07" due to it taking time zones into account. Using `isSame` with 'day' as parameter fixed that, and to me it feels more safe to handle comparisons using a date library.

File: src/utils/dateUtils.ts [hunk 2]
```diff
@@ -86,7 +86,7 @@ export function isInFuture(datestring: string): boolean {
 }
 
 export function isSameDate(first: Date, second: Date): boolean {
-  return first.toISOString().slice(0, 10) == second.toISOString().slice(0, 10);
+  return dayjs(first).isSame(dayjs(second), 'day');
```
[the remark this answers is not in the document]
Comment (ziggabyte) [reply src/utils/dateUtils.ts:89]: I think it makes a lot of sense like you say to let a library handle the comparisons of dates! I did not know of the `endOf` and `startOf` functions in dayjs that you used in the other place, very convenient!

**Review by ziggabyte — CHANGES_REQUESTED** [review]
Nice with the messages and the work on the dates! Almost there!

I noticed another small thing when playing with it - the little colour dot should be blue as long as the email is scheduled (not sent) and green only after it has been sent. Now it is green on the day it is going to be sent, even if it is not sent already. 

bild

would you like to investigate this and see how this could be changed so we get the right colour for before and after being sent?

**Review by ziggabyte — APPROVED** [review]
Great job with this! Just to make sure it would make it into the next release I took on the task of fixing the colors of the status dots (which is a deeper issue that needs more harmonizing in the project in general - but, that's for another day). 

So nice to have you contributing on the project when the time and energy allows :D

**Comment by Jrende** [comment]
I realized that I forgot about this ticket, thank you for doing the final step!

**Pull request merged**